src/mod_extdisco.erl

```diff
@@ -136,3 +137,6 @@ filter_undefined_elements(Elements) ->
     lists:filter(fun({_, undefined}) -> false;
                     (_)              -> true
                  end, Elements).
+
+i2b(X) when is_integer(X) -> integer_to_binary(X);
+i2b(X) -> X.
```

Comment:
I changed the configuration format as well. To be consistent with the other parts of code I changed the port to be an int instead of string. 
This one is just in case the port option is not there, when `Port = gen_mod:get_opt(port, Opts, undefined)` would return atom `undefined`, which we like and want to leave there as we filter out all undefined options later on.

Reply:
Aaaaaaah the `undefined` case, I missed that one. Clear now 🤷‍♂️ 👍